src/dvclive/studio.py

```diff
@@ -0,0 +1,72 @@
+from os import getenv
+
+from dvclive.env import STUDIO_ENDPOINT
+from dvclive.utils import parse_scalars
+
+
+def _get_unsent_datapoints(plot, latest_step):
+    return [x for x in plot if int(x["step"]) >= latest_step]
+
+
+def _cast_to_numbers(datapoints):
+    for datapoint in datapoints:
+        for k, v in datapoint.items():
+            if k == "step":
+                datapoint[k] = int(v)
+            elif k == "timestamp":
+                continue
+            else:
+                datapoint[k] = float(v)
+    return datapoints
+
+
+def _to_dvc_format(plots):
+    formatted = {}
+    for k, v in plots.items():
+        formatted[k] = {"data": v}
+    return formatted
+
+
+def _get_updates(live):
+    plots, metrics = parse_scalars(live)
+    latest_step = live._latest_studio_step  # pylint: disable=protected-access
+
+    for name, plot in plots.items():
+        datapoints = _get_unsent_datapoints(plot, latest_step)
+        plots[name] = _cast_to_numbers(datapoints)
+
+    metrics = {live.summary_path: {"data": metrics}}
+    plots = _to_dvc_format(plots)
+    return metrics, plots
+
+
+def post_to_studio(live, event_type):
+    import requests
+    from requests.exceptions import Timeout
+
+    data = {
+        "type": event_type,
+        "repo_url": live.studio_url,
+        "rev": live.rev,
+        "client": "dvclive",
+    }
+
+    if event_type == "data":
+        metrics, plots = _get_updates(live)
+        data["metrics"] = metrics
+        data["plots"] = plots
+        data["step"] = live.get_step()
+
+    try:
+        response = requests.post(
+            getenv(STUDIO_ENDPOINT, "https://studio.iterative.ai/api/live"),
+            json=data,
+            headers={
+                "Content-type": "application/json",
+                "Authorization": f"token {live.studio_token}",
+            },
+            timeout=5,
+        )
+    except Timeout:
```

Comment:
Might be worth to also catch  `requests.exceptions.ConnectionError`

Reply:
Used the base class `RequestException`